### PR TITLE
6 creds

### PR DESCRIPTION
**Pull request by loadzero**
Handle credentials and url override options.

Setting the credfile option will override the path that the kinesis library uses to grab the creds.
If it is NULL, the default behaviour occurs, which is to look in environment variables and the home directory.

endpoint override is useful for testing with things like kinesalite.

### CODE REVIEW

File: kinesis_consumer.cpp [hunk 1]
```diff
@@ -148,6 +149,28 @@ lookup_region(const char *r)
 	return -1;
 }
 
+static void
+parse_url(ClientConfiguration &config,
```
Comment (usmanm) [inline kinesis_consumer.cpp:153]: Use some built in text search parser. This will fail for malformed URLs.

**Comment by usmanm** [comment]
Looks good except for the URL parsing stuff.

File: kinesis_consumer.cpp (hunk 1 above)
Comment (loadzero) [inline kinesis_consumer.cpp:153]: I can't see any hard failure here. As written it is meant to pass on the constituent parts to the next module that uses them.

Are you referring to that it accepts malformed URLs?

Comment (usmanm) [inline kinesis_consumer.cpp:153]: I'm not sure how the AWS library will respond. I meant we should add some validation for the URL. Feel free to to in a separate diff.

Comment (loadzero) [inline kinesis_consumer.cpp:153]: No worries - They do have a built-it URI parser, it doesn't do any validation though. The URL gets handled down to curl and bubbles back up as a curl error. I concur that a more standard parse and validate would be nice though - TODO

**Pull request closed without merging**